apps/frontend/src/hooks/useWorkoutState.ts

```diff
@@ -0,0 +1,111 @@
+import { useLiveQuery } from 'dexie-react-hooks';
+import {
+  db,
+  defaultWorkoutState,
+  initWorkoutstate,
+  startNewWorkout,
+} from '../db';
+import { useEffect, useState } from 'react';
+import { millisToHoursMinutesAndSeconds } from '@dundring/utils';
+
+let hasPreCalculated = false;
+export const numberOfGraphDataPoints = 500;
+export const useWorkoutState = () => {
+  const [showRecoverPrompt, setShowRecoverPrompt] = useState(true);
+
+  const state =
+    useLiveQuery(() => db.workoutState.limit(1).last()) ?? defaultWorkoutState;
+
+  const trackedData =
+    useLiveQuery(
+      () =>
+        db.workoutDataPoint
+          .where('workoutNumber')
+          .equals(state.workoutNumber)
+          .reverse()
+          .toArray(),
+      [state.workoutNumber]
+    )
+      ?.toReversed()
```

Comment:
are these reverses canceling each other?
if not it maybe needs some comment (or i am missing something obvious)

Reply:
same for other queries

Reply:
Yeah, I think so – I assume they just were left there after a nasty copypaste from the `graphData` query, where we want to limit the 500 newest datapoints and then reverse the result 🤓 

Will fix!